funding/manager.go

```diff
@@ -3123,55 +3124,120 @@ func (f *Manager) waitForFundingConfirmation(
 	log.Infof("Waiting for funding tx (%v) to reach %v confirmations",
 		txid, numConfs)
 
-	var confDetails *chainntnfs.TxConfirmation
-	var ok bool
+	// Monitor the confirmation event for the funding transaction. And
+	// return when the funding tx has received the required number of
+	// confirmations.
+	for {
+		select {
+		case updDetails := <-confNtfn.Updates:
+			log.Tracef("funding tx %s received a confirmation, %d "+
+				" number of confirmations still required", txid,
+				updDetails.NumConfsLeft)
+
+			// Handle first confirmation of the funding transaction.
+			if updDetails.NumConfsLeft == numConfs-1 {
+				log.Infof("funding tx %s received first "+
+					"confirmation", txid)
+
+				err := completeChan.MarkConfirmationHeight(
+					updDetails.BlockHeight,
+				)
+				if err != nil {
+					log.Errorf("failed to mark "+
+						"confirmation height: %v", err)
 
-	// Wait until the specified number of confirmations has been reached,
-	// we get a cancel signal, or the wallet signals a shutdown.
-	select {
-	case confDetails, ok = <-confNtfn.Confirmed:
-		// fallthrough
+					return
+				}
+			}
 
-	case <-cancelChan:
-		log.Warnf("canceled waiting for funding confirmation, "+
-			"stopping funding flow for ChannelPoint(%v)",
-			completeChan.FundingOutpoint)
-		return
+			// If we haven't reached final confirmation, continue
+			// waiting for confirmations.
+			if updDetails.NumConfsLeft > 0 {
+				continue
+			}
 
-	case <-f.quit:
-		log.Warnf("fundingManager shutting down, stopping funding "+
-			"flow for ChannelPoint(%v)",
-			completeChan.FundingOutpoint)
-		return
-	}
+			log.Infof("funding tx %s received all confirmations",
+				txid)
 
-	if !ok {
-		log.Warnf("ChainNotifier shutting down, cannot complete "+
-			"funding flow for ChannelPoint(%v)",
-			completeChan.FundingOutpoint)
-		return
-	}
+			// We've reached final confirmation. Wait for the
+			// confirmation event to trigger and then send the
+			// result to the confirmation channel.
+			err := f.handleFinalConfirmation(
+				confNtfn, confChan, completeChan,
+			)
+			if err != nil {
+				log.Errorf("failed to handle final "+
+					"confirmation: %v", err)
+			}
 
-	fundingPoint := completeChan.FundingOutpoint
-	log.Infof("ChannelPoint(%v) is now active: ChannelID(%v)",
-		fundingPoint, lnwire.NewChanIDFromOutPoint(fundingPoint))
+			return
+
+		case <-confNtfn.NegativeConf:
+			log.Warnf("funding tx %s was reorged out; channel "+
+				"point: %s", txid, completeChan.FundingOutpoint)
+
+			// Reset the confirmation height to 0 because the
+			// funding transaction was reorged out.
+			err := completeChan.MarkConfirmationHeight(uint32(0))
+			if err != nil {
+				log.Errorf("failed to update state for "+
```

Comment:
Nit: instead of a general state use: failed to reset confirmation height for ...